controllers/space/space_controller_test.go

```diff
@@ -729,24 +732,117 @@ func TestPromoteSpace(t *testing.T) {
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 1 * time.Second,
+		}, res) // explicitly requeue while the NSTemplate update is triggered by its controller
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+			Exists().
+			HasTier(basicTier.Name).
+			HasSpecTargetCluster("member-1").
+			HasStatusTargetCluster("member-1").
+			HasConditions(spacetest.Updating()).
+			HasMatchingTierLabelForTier(olderBasicTier)
+		nsTmplSet = nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+			Exists().
+			HasTierName(basicTier.Name).
+			HasClusterResourcesTemplateRef("basic-clusterresources-123456new").
+			HasNamespaceTemplateRefs("basic-code-123456new", "basic-dev-123456new", "basic-stage-123456new").
+			Get()
+		require.True(t, tierutil.TierHashMatches(basicTier, nsTmplSet.Spec))
+
+		t.Run("requeue while NSTemplateSet is not ready", func(t *testing.T) {
+			// given another round of requeue while NSTemplateSet is *not ready*
+			nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
+				nstemplatetsettest.Updating(), // NSTemplateSet status updated
+			}
+			err := member1.Client.Update(context.TODO(), nsTmplSet)
+			require.NoError(t, err)
+
+			// when
+			res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+			// then
+			require.NoError(t, err)
+			assert.False(t, res.Requeue)
+			s := spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+				Exists().
+				HasTier(basicTier.Name).
+				HasSpecTargetCluster("member-1").
+				HasStatusTargetCluster("member-1").
+				HasConditions(spacetest.Updating()).
+				HasMatchingTierLabelForTier(olderBasicTier). // label still not updated
+				Get()
+
+			t.Run("NSTemplateSet is provisioned", func(t *testing.T) {
+				// given another round of requeue with NSTemplateSet now *ready*
+				nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
+					nstemplatetsettest.Provisioned(),
+				}
+				err := member1.Client.Update(context.TODO(), nsTmplSet)
+				require.NoError(t, err)
+
+				// hack: change Space's updating condition timestamp so that it appears the status is ready for >1 second since it's required before can go to Ready condition
+				s.Status.Conditions[0].LastTransitionTime = metav1.NewTime(s.Status.Conditions[0].LastTransitionTime.Time.Add(-1 * time.Second))
+				err = hostClient.Status().Update(context.TODO(), s)
+				require.NoError(t, err)
+
+				// when
+				res, err = ctrl.Reconcile(context.TODO(), requestFor(s))
+
+				// then
+				require.NoError(t, err)
+				assert.Equal(t, reconcile.Result{Requeue: false}, res) // no more requeue.
+				spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+					Exists().
+					HasStatusTargetCluster("member-1").
+					HasConditions(spacetest.Ready()).
+					HasMatchingTierLabelForTier(basicTier). // label updated
+					HasFinalizer()
+			})
+		})
```

Comment:
If you slightly modify the tests then you can easily remove also the `"requeue while NSTemplateSet is not ready"` one - it doesn't verify anything new which wouldn't be already verified earlier, am I correct?
```suggestion
		s = spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
			Exists().
			HasTier(basicTier.Name).
			HasSpecTargetCluster("member-1").
			HasStatusTargetCluster("member-1").
			HasConditions(spacetest.Updating()).
			HasMatchingTierLabelForTier(olderBasicTier).
			Get()
		nsTmplSet = nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
			Exists().
			HasTierName(basicTier.Name).
			HasClusterResourcesTemplateRef("basic-clusterresources-123456new").
			HasNamespaceTemplateRefs("basic-code-123456new", "basic-dev-123456new", "basic-stage-123456new").
			Get()
		require.True(t, tierutil.TierHashMatches(basicTier, nsTmplSet.Spec))

		t.Run("NSTemplateSet is provisioned", func(t *testing.T) {
			// given another round of requeue with NSTemplateSet now *ready*
			nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
				nstemplatetsettest.Provisioned(),
			}
			err := member1.Client.Update(context.TODO(), nsTmplSet)
			require.NoError(t, err)
			// hack: change Space's updating condition timestamp so that it appears the status is ready for >1 second since it's required before can go to Ready condition
			s.Status.Conditions[0].LastTransitionTime = metav1.NewTime(s.Status.Conditions[0].LastTransitionTime.Time.Add(-1 * time.Second))
			err = hostClient.Status().Update(context.TODO(), s)
			require.NoError(t, err)

			// when
			res, err = ctrl.Reconcile(context.TODO(), requestFor(s))

			// then
			require.NoError(t, err)
			assert.Equal(t, reconcile.Result{Requeue: false}, res) // no more requeue.
			spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
				Exists().
				HasStatusTargetCluster("member-1").
				HasConditions(spacetest.Ready()).
				HasMatchingTierLabelForTier(basicTier). // label updated
				HasFinalizer()
		})
```

Reply:
Well actually, I thought of it as verifying the Space status when the NSTemplateSet status changes to Updating. Maybe it's worth renaming the test to `NSTemplateSet status changes to Updating` and its subtest to `NSTemplateSet status changes to Provisioned` for clarity instead?

Reply:
But what is the new functionality that is tested here? It's already covered in the previous "promotion" test, right?

Reply:
Ok, that's a fair point. Since it's covered in the promotion test then I guess we can remove it here.